local_planner/src/local_footstep_planner.cpp

```diff
@@ -174,47 +171,67 @@ void LocalFootstepPlanner::computeFootPlan(
             body_vel_touchdown, ref_body_vel_touchdown, body_ang_vel_touchdown,
             ref_body_ang_vel_touchdown, grf_midstance;
 
-        // Compute the horizon index of midstance
-        int midstance = i + half_duty_cycle;
-
-        // Compute body pose at midstance from either the trajectory or Raibert
-        // heuristic directly
-        if (midstance < horizon_length_) {
-          body_pos_midstance = body_plan.block<1, 3>(midstance, 0);
-          body_rpy_midstance = body_plan.block<1, 3>(midstance, 3);
-          // grf_midstance = grf_plan.block<1, 3>(midstance, 3 * j);
+        // Compute the number of timesteps corresponding to the end of stance
+        // phase
+        int end_of_stance = i + period_ * duty_cycles_[j];
+
+        // Compute the body plan including the stance phase
+        Eigen::MatrixXd body_plan_stance;
+        if (end_of_stance > horizon_length_ - 1) {
+          // Integrate the plan if out of the horizon
+          body_plan_stance = Eigen::MatrixXd(end_of_stance + 1, 12);
+          body_plan_stance.topRows(horizon_length_) = body_plan;
+          for (size_t k = horizon_length_; k < end_of_stance; k++) {
+            body_plan_stance.row(k) = computeFutureBodyPlan(
+                k - (horizon_length_ - 1), body_plan.row(horizon_length_ - 1));
+          }
         } else {
-          body_pos_midstance = body_plan.block<1, 3>(horizon_length_ - 1, 0) +
-                               body_plan.block<1, 3>(horizon_length_ - 1, 6) *
-                                   (midstance - (horizon_length_ - 1)) * dt_;
-          body_rpy_midstance = body_plan.block<1, 3>(horizon_length_ - 1, 3);
-          // grf_midstance = grf_plan.block<1, 3>(horizon_length_ - 1, 3 * j);
+          body_plan_stance = body_plan;
         }
 
+        // Compute the hip position during stance
+        std::vector<Eigen::Vector2d> P, R;
+        for (size_t k = i; k < end_of_stance; k++) {
+          quadKD_->worldToNominalHipFKWorldFrame(
+              j, body_plan_stance.row(k).segment(0, 3),
+              body_plan_stance.row(k).segment(3, 3), hip_position_midstance);
+
+          Eigen::Vector2d p;
+          p << hip_position_midstance.x(), hip_position_midstance.y();
+
+          bool duplicate = false;
+          for (size_t l = 0; l < P.size(); l++) {
+            if ((P.at(l) - p).norm() < 1e-3) {
+              duplicate = true;
+              break;
+            }
+          }
```

Comment:
What is the reason for doing this?

Reply:
The minimal circle algorithm cannot accept repeated inputs.